### ISSUE TITLE
Task: Add support Distributed Async Sign API during "Metadata Update"

### ISSUE THREAD

**Issue by kairoaraujo** [repository-service-tuf/repository-service-tuf-api#386]
### What is the task about?

This task is about adding an API to support metadata signatures.

This API will get the signature status and post signature tasks to the RSTUF broker. These tasks are computed by RSTUF Worker(s)

- It depends on #355, which implements the base operations.

#### Method `GET`

- Removes the bootstrap check to support also the "Metadata Update" process
- Get a list of existent `{ROLE}_SIGNING` from RSTUF Redis (settings Repository)
   The `{ROLE}_SIGNING` can be `None` when there is no pending signatures, or contains the current metadata to add the signature
   For example, `{ROLE}_SIGNING`. This key can be `None`, meaning no signature pending or contains the current metadata.
   If it is `None`, it will not be included.
   
   It will retrieve a body as:
  ```json
  {
     "metadata": {
           "<role name>": {"<metadata>"}
     }
  }
  ```
 
#### Method `POST`

- Remove the verify if the bootstrap is pending signature(s) (`BOOTSTRAP` is `signing-<taskid>`)
- Send the [signature](https://theupdateframework.github.io/specification/latest/#signature) to be included to the Metadata
   
   It will accept a payload like:
```json
  {
    "role": "role name",
    "signature": {
        "keyid": "secure systems lib key id",
        "sig": "signature hash"
    }
  }
```

### Parent feature

- https://github.com/repository-service-tuf/repository-service-tuf/issues/327

### References

- #355 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ivanayov** [comment]
I'd like to work on this

**Comment by MVrachev** [comment]
> ### Endpoint: `/api/v1/metadata/sign`
> #### Method `GET`
> * Removes the bootstrap check to support also the "Metadata Update" process
> * Get a list of existent `{ROLE}_SIGNING` from RSTUF Redis (settings Repository)
>   The `{ROLE}_SIGNING` can be `None` when there is no pending signatures, or contains the current metadata to add the signature
>   For example, `{ROLE}_SIGNING`. This key can be `None`, meaning no signature pending or contains the current metadata.
>   If it is `None`, it will not be included.
>   It will retrieve a body as:
>   ```json
>   {
>      "metadata": {
>            "<role name>": {"<metadata>"}
>      }
>   }
>   ```
> 
> #### Method `POST`
> * Remove the verify if the bootstrap is pending signature(s) (`BOOTSTRAP` is `signing-<taskid>`)
> * Send the [signature](https://theupdateframework.github.io/specification/latest/#signature) to be included to the Metadata
>   It will accept a payload like:
> 
> ```json
>   {
>     "role": "role name",
>     "signature": {
>         "keyid": "secure systems lib key id",
>         "sig": "signature hash"
>     }
>   }
> ```
> 

@kairoaraujo I think this description is misleading.
It describes what we need to achieve `metadata update`, which is already done in pr #397.

Can we update the description with what Ivana has to do actually?

**Comment by MVrachev** [comment]
I just realized that I am not sure what is left to be done on this issue...
The reason is that the RSTUF Worker is the one who makes sure that untrustworthy root won't be used and to make all necessary checks.
Am I correct @kairoaraujo?

**Comment by MVrachev** [comment]
After a discussion with @kairoaraujo and @ivanayov we figured out that there is nothing to add or change in this issue.
This issue was created prior the `bootstrap_state()` refactoring when it did make sense to do some changes.
Also, `POST /api/v1/metadata/sign` and `GET /api/v1/metadata/sign` are implemented in another pr.
